x-pack/docs/en/rest-api/security/bulk-update-api-keys.asciidoc

```diff
@@ -2,4 +2,367 @@
+       "level": 2,
+       "trusted": true,
+       "tags": ["dev", "staging"]
+    }
+  }
+}
+------------------------------------------------------------
+
+Resulting in the following API key information.
+
+[source,console-result]
+--------------------------------------------------
+{
+  "id": "H3_AhoIBA9hmeQJdg7ij",
+  "name": "my-other-api-key",
+  "api_key": "134G4ilmT_uGWXHRfJfXXA",
+  "encoded": "SDNfQWhvSUJBOWhtZVFKZGc3aWo6MTM0RzRpbG1UX3VHV1hIUmZKZlhYQQ=="
+}
+--------------------------------------------------
+// TESTRESPONSE[s/H3_AhoIBA9hmeQJdg7ij/$body.id/]
+// TESTRESPONSE[s/134G4ilmT_uGWXHRfJfXXA/$body.api_key/]
+// TESTRESPONSE[s/SDNfQWhvSUJBOWhtZVFKZGc3aWo6MTM0RzRpbG1UX3VHV1hIUmZKZlhYQQ==/$body.encoded/]
+
+Further, assume that the owner user's permissions are:
+
+[[security-api-bulk-update-api-keys-examples-user-permissions]]
+[source,js]
+--------------------------------------------------
+{
+  "cluster": ["all"],
+  "index": [
+    {
+      "names": ["*"],
+      "privileges": ["all"]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+The following example updates the API keys created above, assigning them new role descriptors and metadata.
+
+[source,console]
+----
+POST /_security/api_key/_bulk_update
+{
+  "ids": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "role_descriptors": {
+    "role-a": {
+      "index": [
+        {
+          "names": ["*"],
+          "privileges": ["write"]
+        }
+      ]
+    }
+  },
+  "metadata": {
+    "environment": {
+       "level": 2,
+       "trusted": true,
+       "tags": ["production"]
+    }
+  }
+}
+----
+// TEST[skip:api key ids not available]
+
+A successful call returns a JSON structure indicating that the API keys were updated:
+
+[source,console-result]
+----
+{
+  "updated": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "noops": []
+}
+----
+
+Both API keys' effective permissions after the update will be the intersection of the supplied role descriptors and the <<security-api-bulk-update-api-keys-examples-user-permissions, owner user's permissions>>:
+
+[source,js]
+--------------------------------------------------
+{
+  "index": [
+    {
+      "names": ["*"],
+      "privileges": ["write"]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+The following example removes the API keys' previously assigned permissions.
+
+[source,console]
+----
+POST /_security/api_key/_bulk_update
+{
+  "ids": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "role_descriptors": {}
+}
+----
+// TEST[skip:api key ids not available]
+
+Which returns the response:
+
+[source,console-result]
+----
+{
+  "updated": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "noops": []
+}
+----
+
+The API keys' effective permissions after the update will the same as the <<security-api-bulk-update-api-keys-examples-user-permissions, owner user's>>:
```

Comment:
Since the owner user's role is list immediately below this, I think we can remove the link here. Clicking the link for the same information does not seem to be a great read experience.